conan/api/subapi/config.py

```diff
@@ -191,3 +192,26 @@ def appending_recursive_dict_update(d, u):
             appending_recursive_dict_update(settings, settings_user)
 
         return Settings(settings)
+
+    def migrate(self):
```

Comment:
Migration is not only config, it is also storage, packages, etc. Maybe it should belong to ConanAPI directly?